packages/hash/frontend/src/components/hooks/useUpdateAuthenticatedUser.ts

```diff
@@ -0,0 +1,117 @@
+import { useMutation } from "@apollo/client";
+import { useCallback, useState } from "react";
+
+import { types } from "@hashintel/hash-shared/types";
+import { extractBaseUri } from "@blockprotocol/type-system-web";
+import { GraphQLError } from "graphql";
+import {
+  UpdatePersistedEntityMutation,
+  UpdatePersistedEntityMutationVariables,
+} from "../../graphql/apiTypes.gen";
+import { mustGetEntity, Subgraph } from "../../lib/subgraph";
+import { AuthenticatedUser, constructAuthenticatedUser } from "../../lib/user";
+import { updatePersistedEntityMutation } from "../../graphql/queries/knowledge/entity.queries";
+import { useAuthenticatedUser } from "./useAuthenticatedUser";
+
+type UpdateAuthenticatedUserParams = {
+  shortname?: string;
+  preferredName?: string;
+};
+
+export const useUpdateAuthenticatedUser = () => {
+  const { authenticatedUser, refetch } = useAuthenticatedUser();
+
+  const [updatePersistedEntity] = useMutation<
+    UpdatePersistedEntityMutation,
+    UpdatePersistedEntityMutationVariables
+  >(updatePersistedEntityMutation, { errorPolicy: "all" });
+
+  const [loading, setLoading] = useState<boolean>(false);
+
+  const updateAuthenticatedUser = useCallback(
+    async (
+      params: UpdateAuthenticatedUserParams,
+    ): Promise<{
+      updatedAuthenticatedUser?: AuthenticatedUser;
+      errors?: readonly GraphQLError[] | undefined;
+    }> => {
+      if (!params.shortname && !params.preferredName) {
+        return { updatedAuthenticatedUser: authenticatedUser };
+      }
+
+      const [
+        {
+          data: { me: latestSubgraph },
+        },
+      ] = await refetch();
+
+      const entityId = latestSubgraph.roots[0]!;
+
+      /** @todo: use a partial update mutation instead */
```

Comment:
Yes, I think this would be really useful. Have [created the task](https://app.asana.com/0/1202805690238892/1203285029221330/f)  and referenced the todo in https://github.com/hashintel/hash/pull/1299/commits/93decc09336dfed4e55281e66d72dc4e68ef25f0

Reply:
Thanks, just to flag though, I've updated the task to capture what I meant.